src/main/scala/com/fulcrumgenomics/util/miseq/SampleSheet.scala

```diff
@@ -0,0 +1,152 @@
+  * this list of conditions and the following disclaimer.
+  *
+  * 2. Redistributions in binary form must reproduce the above copyright notice,
+  * this list of conditions and the following disclaimer in the documentation
+  * and/or other materials provided with the distribution.
+  *
+  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+  * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+  * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+  * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+  * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+  * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+  * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+  * POSSIBILITY OF SUCH DAMAGE.
+  */
+package com.fulcrumgenomics.util.miseq
+
+import java.nio.file.Path
+
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+
+/**
+  * Stores information about samples from an Illumina Experiment Manager Sample Sheet (typically a MiSeq).  The
+  * samples may also include derived information.  If library id is not specified, it will be set to the sample name.
+  *
+  * Optional fields include support for specifying the expected sample barcode for each sample.  The sample barcode
+  * can be present as a sub-sequence (or sub-sequences) in the i7 or i5 read.  If additional bases are found in the i7
+  * or i5 read, such as molecular barcodes, the should be included as Ns.  It is up to the developer to obtain the
+  * correct read structure elsewhere to infer which bases are sample barcode and which bases are not (ex. molecular
+  * identifiers).  If the sample barcode is inline in either read of a pair, the sample barcode sequence can be specified
+  * here.  In this case, only include the sample barcode, and not the additional bases, such as template bases.
+  *
+  * NB: the lookup in the columns in the Sample Sheet is case insensitive.
+  *
+  * @author Nils Homer
+  */
+object SampleSheet {
+  /** Required header names for standard fields in the Illumina Experiment Manager Sample Sheet. */
+  val SampleId: String       = "Sample_ID"
+  val SampleName: String     = "Sample_Name"
+  val LibraryId: String      = "Library_ID"
+  val SampleProject: String  = "Sample_Project"
+  val Description: String    = "Description"
+  /** Optional header names for standard fields in the Illumina Experiment Manager Sample Sheet. */
+  val Lane: String           = "Lane"
+  val R1BarcodeBases: String = "R1_Barcode_Bases"
+  val R2BarcodeBases: String = "R2_Barcode_Bases"
+  val I7Bases: String        = "Index"
+  val I5Bases: String        = "Index2"
```

Comment:
I could break this out to a separate object, but it seems natural to live here, though folks may want to extend that class to add more header names.  Thoughts?